modules/apps/frontend-js/frontend-js-walkthrough-web/test/js/Walkthrough.test.js

```diff
@@ -0,0 +1,251 @@
+
+		cleanUp();
+	});
+
+	it(`when clicking on Previous, it navigates to the given URL of 'previous'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(PAGE_WITH_PREVIOUS_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		userEvents.click(screen.getByText('ok'));
+
+		userEvents.click(screen.getByText('previous'));
+
+		expect(navigate).toBeCalledWith(
+			PAGE_WITH_PREVIOUS_MOCK.steps[1].previous
+		);
+
+		cleanUp();
+	});
+
+	it(`when clicking on Next, without passing 'next', it shows the next Walkthrough without navigating`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(PAGE_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		const okButton = screen.getByText('ok');
+
+		userEvents.click(okButton);
+
+		expect(screen.queryByText('Hello2')).toBeInTheDocument();
+
+		cleanUp();
+	});
+
+	it(`when clicking on Previous, without passing 'previous', it shows the previous Walkthrough without navigating`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(PAGE_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		const okButton = screen.getByText('ok');
+
+		userEvents.click(okButton);
+
+		expect(screen.queryByText('Hello2')).toBeInTheDocument();
+
+		userEvents.click(screen.getByText('previous'));
+
+		expect(screen.queryByText('Hello1')).toBeInTheDocument();
+
+		cleanUp();
+	});
+
+	it('throws an error when the provided selector is not defined', () => {
+		const cleanUp = setupDocument();
+
+		const errorSpy = jest.spyOn(console, 'error').mockImplementation();
+
+		renderWalkthrough(INVALID_NODE_SELECTOR_MOCK);
+
+		expect(errorSpy).toHaveBeenCalled();
+		expect(errorSpy).toHaveBeenCalledTimes(1);
+
+		cleanUp();
+	});
+
+	it(`when 'darkbg' is set to false adds a 'lfr-walkthrough-element-shadow' to the nodeToHighlight'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(BOX_SHADOW_ELEMENT_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		const elementToBeHighlighted = screen.getByTestId(
+			'external-react-tree-button'
+		);
+
+		expect(elementToBeHighlighted).toHaveClass(
+			'lfr-walkthrough-element-shadow'
+		);
+
+		cleanUp();
+	});
+
+	it(`given the following layoutRelativeURLs and the value of '/home' the component needs to be rendered`, () => {
+		const POSSIBLE_LAYOUT_RELATIVE_URLS = [
+			'/undefined',
+			'/es/web/home/undefined',
+			'/es/web/home?redirectURL=/es/web/guest/home/abc',
+			'/home',
+			'/es/home',
+			'/es/web/guest/home',
+			'/home/abc',
+			'/es/home/abc',
+			'/es/web/guest/home/abc',
+			'/home/abc/home',
+			'/es/home/abc/home',
+			'/es/web/guest/home/abc/home',
+		];
+
+		POSSIBLE_LAYOUT_RELATIVE_URLS.forEach((url) => {
```

Comment:
Another option here and perhaps more recommended would be to use [`it.each`](https://jestjs.io/docs/26.x/api#testeachtablename-fn-timeout) instead of a single test with a forEach.

Reply:
Updated! Thanks